Shared/Samples/Show viewshed from geoelement in scene/ShowViewshedFromGeoelementInSceneView.swift

```diff
@@ -0,0 +1,194 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import ArcGIS
+
+struct ShowViewshedFromGeoelementInSceneView: View {
+    /// The view model for the sample.
+    @StateObject private var model = Model()
+    
+    var body: some View {
+        SceneView(scene: model.scene,
+                  cameraController: model.cameraController,
+                  graphicsOverlays: [model.graphicsOverlay],
+                  analysisOverlays: [model.analysisOverlay]
+        )
+        .onSingleTapGesture { _, mapPoint in
+            // Start a timer to animate the tank moving towards the new waypoint.
+            model.waypoint = mapPoint
+            model.animationTimer = Timer.scheduledTimer(withTimeInterval: 0.1, repeats: true) { _ in
+                model.animate()
+            }
```

Comment:
Since the model owns the timer and is responsible for stopping the timer, it makes sense to make it responsible for starting the timer as well, keeping the timer as a private implementation detail of the model. Although, it seems odd for a model to have a timer that is purely for updating the UI, so another option is to move the timer to the view itself. That's probably a better option because the timer should be stopped when the view disappears.